nova_plugin/server.py

```diff
@@ -473,14 +473,16 @@ def get_server_by_context(nova_client):
 
 def _set_network_and_ip_runtime_properties(server):
     ips = {}
-    try:
-        _, default_network_ips = server.networks.items()[0]
-    except IndexError:
+    _, ips_from_server_networks = server.networks.items()
+    if not ips_from_server_networks:
         raise NonRecoverableError(
             'The server was created but not attached to a network. '
             'Cloudify requires that a server is connected to '
             'at least one port.'
         )
+
+    default_network_ips = ips_from_server_networks[0]
```

Comment:
You can still get an IndexError.. shouldn't the if relate to ips_from_server_networks.items() ?